ksql-common/src/main/java/io/confluent/ksql/function/KsqlTableFunction.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.function;
+
+import io.confluent.ksql.schema.ksql.types.SqlType;
+import java.util.List;
+import org.apache.kafka.connect.data.Schema;
+
+public interface KsqlTableFunction<I, O> extends FunctionSignature {
+
+  Schema getReturnType();
+
+  SqlType returnType();
+
+  List<O> flatMap(List<I> input);
```

Comment:
Why are we assuming the input is a `List`???  Isn't that specific to explode?  It could be anything, right? e.g. a `STRING` or `INT`....

Reply:
Will resolve in further PR

Reply:
done